### ISSUE TITLE
Ampersand in Variables

### ISSUE THREAD

**Issue by Shingles007** [steebn/WebFOCUS-VSCode#4]
Hello,

I've noticed that double clicking a variable name will not select the ampersand at the front of the name.  Can this be changed so that the ampersand is included?  I wouldn't have asked for this, but I noticed a related issue, its below...

I sometimes keep line wrapping on, so that I can see the whole line on my screen.  This is especially useful if I am viewing multiple files side by side.  Anyhow, if a variable name needs to get wrapped to the next line, it does not keep the ampersand, so the ampersand is the last character on the previous line, and the new line starts with the text of the variable name (no ampersand).  Can this be changed so that the ampersand is kept with the rest of the variable name (ie they both go to the new line)?

**Comment by steebn** [comment]
I will look into this and see if it is possible.

**Comment by steebn** [comment]
If you go to your vscode settings, there is the following setting:

_"editor.wordSeparators": "`~!@#$%^&*()-=+[{]}\\|;:'\",.<>/?",_

If you customize your setting, and remove the amper from the string of special characters, that fixes the double click issue and selects the amper along with the variable  name. Unfortunately, this doesn't seem to correct the word break when wrapping.

Still looking into this...

**Comment by steebn** [comment]
After digging into the VSCode GitHub Repo, I found a setting called _wordWrapBreakAfterCharacters_ that seems to be hidden, and has the amper included. [https://github.com/Microsoft/vscode/blob/17454d4e88886404af130639b979498b227a9167/src/vs/editor/common/config/editorOptions.ts#L2523](url)

I added this setting in my VS Code User settings and removed the amper from it, and it shows as an Unknown Configuration Setting with a green squiggly line under it. However, with that modified "unknown" setting in place, it corrects the issue of wrapping that you explained.

So, I have the following settings in place in my user settings json (you will need to make sure that your workbench>Settings Editor>Editor is set to json not ui) and it corrects the issues that you explained.

    "editor.wordWrap": "on",
    "editor.wordSeparators": "`~!@#%^*()-=+[{]}\\|;:'\",.<>/?",
    "editor.wordWrapBreakAfterCharacters": " \t})]?|,;¢°′″‰℃、。｡､￠，．：；？！％・･ゝゞヽヾーァィゥェォッャュョヮヵヶぁぃぅぇぉっゃゅょゎゕゖㇰㇱㇲㇳㇴㇵㇶㇷㇸㇹㇺㇻㇼㇽㇾㇿ々〻ｧｨｩｪｫｬｭｮｯｰ”〉》」』】〕）］｝｣",

I will see if I can incorporate these into the extension, but this can be a workaround for you in the meantime.
